source/crud/tests/unified/bulkWrite-replaceOne-dots_and_dollars.yml

```diff
@@ -0,0 +1,165 @@
+description: "bulkWrite-replaceOne-dots_and_dollars"
+
+schemaVersion: "1.0"
+
+createEntities:
+  - client:
+      id: &client0 client0
+      observeEvents: [ commandStartedEvent ]
+  - database:
+      id: &database0 database0
+      client: *client0
+      databaseName: &database0Name crud-tests
+  - collection:
+      id: &collection0 collection0
+      database: *database0
+      collectionName: &collection0Name coll0
+
+initialData: &initialData
+  - collectionName: *collection0Name
+    databaseName: *database0Name
+    documents:
+      - { _id: 1 }
+
+tests:
+  - description: "Replacing document with top-level dotted key on 3.6+ server"
+    runOnRequirements:
+      - minServerVersion: "3.6"
+    operations:
+      - name: bulkWrite
+        object: *collection0
+        arguments:
+          requests:
+            - replaceOne:
+                filter: { _id: 1 }
+                replacement: &dottedKey { _id: 1, a.b: 1 }
+        expectResult: &bulkWriteResult
+          deletedCount: 0
+          insertedCount: 0
+          insertedIds: { $$unsetOrMatches: { } }
+          matchedCount: 1
+          modifiedCount: 1
+          upsertedCount: 0
+          upsertedIds: { }
+    expectEvents: &expectEventsDottedKey
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                update: *collection0Name
+                updates:
+                  - q: { _id: 1 }
+                    u: *dottedKey
+                    multi: { $$unsetOrMatches: false }
+                    upsert: { $$unsetOrMatches: false }
+    outcome:
+      - collectionName: *collection0Name
+        databaseName: *database0Name
+        documents:
+          - *dottedKey
+
+  - description: "Replacing document with top-level dotted key on pre-3.6 server yields server-side error"
+    runOnRequirements:
+      - maxServerVersion: "3.4.99"
```

Comment:
This is intentional. 3.5.x is the development release series for what ultimately becomes 3.6.x, so I'm allowing this to run on any stable release series before that. In practice, this makes little difference since a driver is unlikely to ever test on development versions before the latest release (e.g. 4.3.x became irrelevant as soon as 4.4.0 was tagged).